sqlglot/dialects/oracle.py

```diff
@@ -282,3 +282,181 @@ def add_column_sql(self, expression: exp.AlterTable) -> str:
             if len(expression.args.get("actions", [])) > 1:
                 return f"ADD ({actions})"
             return f"ADD {actions}"
+
+
+def eliminate_join_marks(ast: exp.Expression) -> exp.Expression:
```

Comment:
i think this could perhaps be called in the parser. once everything is done, if we've parsed a join_mark (which happens in _parse_column),

we can call this method to transform and remove the join marker.

if we can handle all cases of join marker elimination, maybe we can get rid of the whole concept?